packages/office-ui-fabric-react/src/components/Button/Button.test.tsx

```diff
@@ -113,141 +113,209 @@ describe('Button', () => {
-        >
-          Hello
-        </DefaultButton>
-      );
-      renderedDOM = ReactDOM.findDOMNode(button as React.ReactInstance);
-      expect(renderedDOM.getAttribute('aria-pressed')).toEqual('false');
+        expect(renderedDOM.getAttribute('aria-label')).toEqual('MyLabel');
+        expect(renderedDOM.getAttribute('aria-labelledby')).toBeNull();
+        expect(renderedDOM.getAttribute('aria-describedby')).toBeNull();
+      });
 
-      button = ReactTestUtils.renderIntoDocument<any>(
-        <DefaultButton
-          toggle={true}
-          checked={true}
-          split={true}
-          onClick={alertClicked}
-          menuProps={{
-            items: [
-              {
-                key: 'emailMessage',
-                text: 'Email message',
-                iconProps: { iconName: 'Mail' }
-              }
-            ]
-          }}
-        >
-          Hello
-        </DefaultButton>
-      );
-      renderedDOM = ReactDOM.findDOMNode(button as React.ReactInstance);
-      expect(renderedDOM.getAttribute('aria-pressed')).toEqual('true');
+      it('applies aria-labelledby', () => {
+        const button: any = ReactTestUtils.renderIntoDocument<any>(
+          <DefaultButton href="http://www.microsoft.com" target="_blank" aria-labelledby="someid">
+            Hello
+          </DefaultButton>
+        );
+        const renderedDOM: any = ReactDOM.findDOMNode(button as React.ReactInstance);
+
+        expect(renderedDOM.getAttribute('aria-labelledby')).toEqual('someid');
+        expect(renderedDOM.getAttribute('aria-describedby')).toBeNull();
+      });
+
+      it('does not apply aria-labelledby to a button with no text', () => {
+        const button: any = ReactTestUtils.renderIntoDocument<any>(
+          <DefaultButton href="http://www.microsoft.com" target="_blank" aria-describedby="someid" />
+        );
+        const renderedDOM: any = ReactDOM.findDOMNode(button as React.ReactInstance);
+
+        expect(renderedDOM.getAttribute('aria-labelledby')).toBeNull();
+        expect(renderedDOM.getAttribute('aria-describedby')).toEqual('someid');
+      });
+
+      it('applies aria-labelledby and aria-describedby', () => {
+        const button: any = ReactTestUtils.renderIntoDocument<any>(
+          <DefaultButton
+            href="http://www.microsoft.com"
+            target="_blank"
+            ariaDescription="This description is not visible"
+            styles={{ screenReaderText: 'some-screenreader-class' }}
+          >
+            Hello
+          </DefaultButton>
+        );
+        const renderedDOM: any = ReactDOM.findDOMNode(button as React.ReactInstance);
+
+        expect(renderedDOM.getAttribute('aria-label')).toBeNull();
+        expect(renderedDOM.getAttribute('aria-labelledby')).toEqual(renderedDOM.querySelector(`.ms-Button-label`).id);
+        expect(renderedDOM.getAttribute('aria-describedby')).toEqual(
+          renderedDOM.querySelector('.some-screenreader-class').id
+        );
```

Comment:
For all of these value comparison tests, would it be too pedantic to worry that both values might be undefined, causing a false positive?